### PR TITLE
Improve System.build_info/0 spec

### PR DESCRIPTION
**Pull request by eksperimental**
Be specific about map fields.

### CODE REVIEW

File: lib/elixir/lib/system.ex [hunk 1]
```diff
@@ -205,7 +205,13 @@ defmodule System do
...
 
   """
-  @spec build_info() :: %{required(atom) => String.t()}
+  @spec build_info() :: %{
+          build: String.t(),
+          date: String.t(),
+          revision: String.t(),
+          version: String.t(),
+          otp_release: String.t()
+        }
```
Comment (ericmj) [inline lib/elixir/lib/system.ex]: Actually, maybe we should define this as a type that we refer to from this spec?

Comment (eksperimental) [reply lib/elixir/lib/system.ex]: sorry, I don't get it

Comment (ericmj) [reply lib/elixir/lib/system.ex]: Like this. Not sure what's preferable, let's wait for other comments.

```elixir
@type build_info() :: %{...}
@spec build_info() :: build_info()
```

Comment (eksperimental) [reply lib/elixir/lib/system.ex]: I see. Well, if we are not calling it from any other function, I think we shouldn't yet

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**